autorest/codegen/serializers/model_base_serializer.py

```diff
@@ -74,8 +77,12 @@ def get_properties_to_initialize(model: ObjectSchema) -> List[Property]:
         return properties_to_initialize
 
     @staticmethod
-    def declare_model(model: ObjectSchema) -> str:
-        basename = "msrest.serialization.Model"
+    def declare_model(code_model: CodeModel, model: ObjectSchema) -> str:
```

Comment:
you shouldn't ned to add the code model as a parameter. If you make this method a method on the class (get rid of the staticmethod decorator and add self to the signature), you'll have access to code model already